lib/puppet/util/run_mode.rb

```diff
@@ -55,11 +55,22 @@ def which_dir( system, user )
 
     class UnixRunMode < RunMode
       def conf_dir
-        which_dir("/etc/puppet", "~/.puppet")
+        which_dir("/etc/puppetlabs/agent", "~/.puppet")
       end
 
       def var_dir
-        which_dir("/var/lib/puppet", "~/.puppet/var")
+        # If Puppet is run as a non-root user and vardir is specified via the commandline,
+        # we'll need to use this user defined value when creating paths for rundir and logdir
+        user_vardir = if Puppet[:vardir] then Puppet[:vardir] else "~/.puppet/var" end
```

Comment:
@whopper Style-wise, this is weird. Why not `user_vardir = Puppet[:vardir] || "~/.puppet/var"` or `user_vardir = Puppet[:vardir] ? Puppet[:vardir] : "~/.puppet/var"`? Not a blocker, just weird.